packages/model-viewer/src/features/scene-graph/model.ts

```diff
@@ -245,4 +262,109 @@ export class Model implements ModelInterface {
     }
     await Promise.all(promises);
   }
+
+  [$cloneMaterial](index: number, newMaterialName: string): Material {
+    const material = this.materials[index];
+
+    if (!material.isLoaded) {
+      console.error(`Cloning an unloaded material,
+           call 'material.ensureLoaded() before cloning the material.`);
```

Comment:
Correct just trying to avoid making it async as that generate a ton of changes in material_panel, which might be ok I just wouldn't want to do that in this PR.